main.py

```diff
@@ -52,3 +59,50 @@ def add_product_to_db(product: ProductBase):
     new_product = app.state.mongo_collection.find_one({'_id': inserted_id})
     # Return the new product as a dictionary
     return new_product
+
+
+def relativedelta(months):
+    pass
+
+
+@app.get("/inflation")
+def calculate_inflation(type_product: str = "all", start_date: datetime = datetime.now(), end_date: datetime = datetime.now()):
+
+    """
+    Calculates the total sum of product prices within a specified period.
+
+    Args:
+        type_product (str, optional): The type of product to filter (default is "all").
+        start_date (datetime): The start date of the period.
+        end_date (datetime): The end date of the period.
+
+    Returns:
+        float: The total sum of product prices within the specified period.
+    """
+    products_current_period = collection.find()
+
+    total_sum_current_period = 0
+
+    for product in products_current_period:
+        price = product["Price"]
+        creation_date = datetime.strptime(product["Creation Date"], "%d.%m.%Y")
+
+        # Filter out products outside the specified period
+        if start_date <= creation_date <= end_date:
+            total_sum_current_period += price
+
+    products_previous_period = collection.find()
+
+    total_sum_previous_period = 0
+
+    for product in products_previous_period:
+        price = product["Price"]
+        creation_date = datetime.strptime(product["Creation Date"], "%d.%m.%Y")
+
+        # Filter out products for the previous period
+        previous_start_date = start_date - relativedelta(months=PREVIOUS_PERIOD_MONTHS)
+        previous_end_date = end_date - relativedelta(months=PREVIOUS_PERIOD_MONTHS)
+        if previous_start_date <= creation_date <= previous_end_date:
+            total_sum_previous_period += price
+
+    return total_sum_current_period, total_sum_previous_period
```

Comment:
I know you didn't run your code. How do you want it to work if you don't even run it?